R/evaldyntracer.R

```diff
@@ -1,10 +1,17 @@
-create_dyntracer <- function() .Call(C_create_dyntracer)
+create_dyntracer <- function(base_eval, analysis_dirpath,
```

Comment:
If we are storing R expressions in CSV files then `,` can confuse the parser as R expression can also have a `,`. In that case we may have to use a different `column_separator`. Similar reasoning may apply for `row_separator`. Hence I generalized `CsvWriter` to a `TableWriter`.

If we don't make them constructor arguments, we will at least make them `const std::string` somewhere. In any case, we will prefer referring to them by some name in the source code rather than hard coding them in the implementation. I just took a step further and made it configurable. I don't think it adds any real complexity. It just makes it easy to switch separators if we ever have to.

The plan is to have a few rows (~ 20 - 30) per CSV file in most cases. We have to pre summarize data in the tracer itself. Generally CSV files will work well for this. CSV files will suffice for the standard use cases. We can do something different for non-standard use case. I am strictly against any binary formats because they take away debugging capability. We can always evaluate our choices as we go along.

Reply:
You don't want to escape the values since that will be expensive?